### ISSUE TITLE
Terminating `sharedStdIn` is problematic for commands that call other commands

### ISSUE THREAD

**Issue by kenzieschmoll** [flutter/devtools#6624]
We call `sharedStdIn.terminate()` from the `DevToolsCommandRunner.runCommand` override here: https://github.com/flutter/devtools/blob/master/tool/lib/devtools_command_runner.dart/#L45. This is problematic for commands that have subcommands and commands that call other commands like `update-version` and `release-helper`.
@DanTup

**Comment by DanTup** [comment]
Doh! Is there any good hook to have it always called at the top level without having to code it into each command that might execute processes forwarding the streams, or should I just revert that change?

**Comment by kenzieschmoll** [comment]
Could we check if the command is a subcommand or is contained in a parent process?

**Comment by DanTup** [comment]
I couldn't find a way to do that inside the command (I don't seem to be able to tell if we're the top level), but I realised we could do this inside the entry point.

I've opened https://github.com/flutter/devtools/pull/6629 and confirmed I no longer see "Bad state: Stdin has already been terminated." when running `devtools_tool update-version auto -t dev`.

Let me know if you think there are still issues (and if so, what they are specifically - I don't know if there may be any differences between macOS and Windows here). Thanks!
